File: app/schemas/discussion.py
"""
schemas/discussion.py — Property discussion / Q&A schemas
"""

from typing import Optional, List
from pydantic import BaseModel


class ReplyResponse(BaseModel):
    id: str
    user_id: str
    user_name: str
    message: str
    created_at: str


class DiscussionResponse(BaseModel):
    id: str
    property_id: str
    user_id: str
    user_name: str
    message: str
    created_at: str
    replies: List[ReplyResponse] = []


class DiscussionCreate(BaseModel):
    message: str
    parent_id: Optional[str] = None

File: android/app/src/main/kotlin/com/realestate/app/data/models/Discussion.kt
package com.realestate.app.data.models

import com.google.gson.annotations.SerializedName

data class Reply(
    @SerializedName("id") val id: String = "",
    @SerializedName("user_id") val userId: String = "",
    @SerializedName("user_name") val userName: String = "",
    @SerializedName("message") val message: String = "",
    @SerializedName("created_at") val createdAt: String = ""
)

data class Discussion(
    @SerializedName("id") val id: String = "",
    @SerializedName("property_id") val propertyId: String = "",
    @SerializedName("user_id") val userId: String = "",
    @SerializedName("user_name") val userName: String = "",
    @SerializedName("message") val message: String = "",
    @SerializedName("created_at") val createdAt: String = "",
    @SerializedName("replies") val replies: List<Reply> = emptyList()
)

data class DiscussionCreateRequest(
    @SerializedName("message") val message: String,
    @SerializedName("parent_id") val parentId: String? = null
)
